File: send.rb
#!/usr/bin/ruby 
require 'socket'

if ARGV.size != 1 && ARGV.size != 2
	print "Usage : send.rb host port\n"
	exit
end
#host = gethostbyname(ARGV[0])
host = ARGV[0]
port = ARGV.size == 2 ? ARGV[1].to_i : 1818


# ruby bug...
if "localhost" == host
	host = "127.0.0.1"
end

s = Socket.new(Socket::Constants::AF_INET, Socket::Constants::SOCK_STREAM, 0)
s.connect( Socket.pack_sockaddr_in(port, host))
s.binmode
while ! STDIN.eof do
	begin
		s.write(STDIN.readpartial(65536))
	rescue EOFError 
		s.close
		exit
	end
end

File: send.pl
#!/usr/bin/perl
use Socket;
if ($#ARGV == -1) { print "Usage : send.pl host port\n";exit;}
my $host = $ARGV[0];
my $proto = getprotobyname('tcp');
my $port = $#ARGV > 0 ? $ARGV[1] : 1818;
$handle = &Connect2Server($host, $port, 2);
while (<STDIN>) {
        print $handle $_;
}
close $handle;
exit;

sub interrupt {
        print "interrupted\n";
        exit;
}
sub Connect2Server {
        my ($host, $port) = @_;
        my $find = 0;
        my $sin;

        socket(Socket_Handle, PF_INET, SOCK_STREAM, $proto) || die "Socket: $!";
        $sin = sockaddr_in($port, inet_aton($host));

        select(Socket_Handle);
        $| = 1;
        select(stdout);

        $result = connect(Socket_Handle, $sin);
        if ($result != 1) {
                print "Connection Fail!!\n";
                exit;
        }
        return Socket_Handle;
}
